test/partial-result-stream.ts

```diff
@@ -251,7 +253,63 @@ describe('PartialResultStream', () => {
       );
     });
 
-    it('should resume if there was an error', done => {
+    it('should retry if the initial call returned a retryable error', done => {
+      // This test will emit two rows total:
+      // - UNAVAILABLE error (should retry)
+      // - Two rows
+      // - Confirm all rows were received.
+      const fakeCheckpointStream = through.obj();
+      // tslint:disable-next-line no-any
+      const resetStub = ((fakeCheckpointStream as any).reset = () => {});
+      sandbox.stub(checkpointStream, 'obj').returns(fakeCheckpointStream);
+
+      const firstFakeRequestStream = through.obj();
+      const secondFakeRequestStream = through.obj();
+
+      const requestFnStub = sandbox.stub();
+
+      requestFnStub.onCall(0).callsFake(() => {
+        setTimeout(() => {
+          // This causes a new request stream to be created.
+          firstFakeRequestStream.emit('error', {
+            code: status.UNAVAILABLE,
+            message: 'Error.',
+          } as ServiceError);
+          firstFakeRequestStream.end();
```

Comment:
In order to simulate the real case as we discussed, I think we should remove this. I tried it and it works fine.

Reply:
Done